### PR TITLE
Update IamArn pattern

### PR DESCRIPTION
**Pull request by liumofei-amazon**
*Issue #, if available:*

*Description of changes:*

Update IamArn pattern
Add encryption to SNS topic
Handle edge case in ListHandler
Fix failing contract tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mirelap-amazon — APPROVED** [review]

File: aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/DeleteHandler.java [hunk 1]
```diff
@@ -39,10 +40,10 @@
 
             logger.log(String.format("%s [%s] for accountId [%s] has been successfully deleted!", ResourceModel.TYPE_NAME, profilingGroupName, awsAccountId));
 
-            return ProgressEvent.defaultSuccessHandler(model);
-
+            // Delete handler should not return a model in case of success, according to https://w.amazon.com/bin/view/AWS21/Design/Uluru/ContractTests
```
Comment (mirelap-amazon) [inline aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/DeleteHandler.java]: Please remove this.

Comment (liumofei-amazon) [reply aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/DeleteHandler.java]: Removed all internal links

**Review by mirelap-amazon — APPROVED** [review]
